app/javascript/components/toolbar/Toolbar.jsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+import { ToolbarView } from './ToolbarView';
+import { ToolbarButton } from './ToolbarButton';
+import { ToolbarList } from './ToolbarList';
+import CountContext from './ToolbarContext';
+
+export const ButtonType = {
+  BUTTON: 'button',
+  BUTTON_TWO_STATE: 'buttonTwoState',
+  BUTTON_SELECT: 'buttonSelect',
+  SEPARATOR: 'separator',
+  KEBAB: 'kebab',
+};
+
+const isButton = (item) => (item.type === ButtonType.BUTTON);
+const isButtonTwoState = (item) => (item.type === ButtonType.BUTTON_TWO_STATE);
+const isButtonSelect = (item) => (item.type === ButtonType.BUTTON_SELECT);
+const isKebabMenu = (item) => (item.type === ButtonType.KEBAB);
+
+const isButtonSelectVisible = (props) => props.items.filter((i) => !i.hidden).length !== 0;
+
+const isVisibleButtonOrSelect = (item) => (
+  item.type && (
+    (isButtonSelect(item) && isButtonSelectVisible(item))
+    || (isButton(item) && !item.hidden)
+    || (isButtonTwoState(item) && !item.hidden)
+    || (isKebabMenu(item) && !item.hidden)
+  )
+);
+
+const toolbarGroupHasContent = (group) =>
+  group
+    && group.filter((item) => item
+      && isVisibleButtonOrSelect(item)).length !== 0;
+
+const buttonCase = (item, index, onClick) => {
+  if (isButton(item) || isButtonTwoState(item)) {
+    return <ToolbarButton key={index} {...item} onClick={onClick} />;
+  } if (isButtonSelect(item) && (item.items.length > 0)) {
+    return <ToolbarList key={index} {...item} onClick={onClick} />;
+  }
+  return null;
```

Comment:
Can we make these conditions nicer?

Reply:
@skateman yes it is just moved from old code, In this pr, i moved component from react repo to here as it is needed, changed PF components to CC and fix accessibility, unless there as a performance issue because of the way this code is written,  i don't want to make changes to existing working code. especially, for testing purposes, i am not really sure how and where this code is going to take an affect in ui, so want to keep the changes minimal and make it working.